eos/utils/szego-polynomial.hh

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/utils/power_of.hh>
+
+#include <array>
+
+namespace eos
+{
+	template <unsigned order_>
+    class SzegoPolynomial
+    {
+		private:
+			const double _norm_measure;
+
+			const std::array<double, order_> _verblunsky_coefficients;
+
+			const std::array<double, order_ + 1> _norms;
+
+			std::array<double, order_ + 1> _calculate_norms() const
+			{
+				std::array<double, order_ + 1> result;
+
+				result[0] = _norm_measure;
+
+				for (unsigned i = 1 ; i <= order_ ; ++i)
+				{
+					result[i] = result[i - 1] * (1.0 - power_of<2>(_verblunsky_coefficients[i - 1]));
+				}
+
+				for (unsigned i = 0 ; i <= order_ ; ++i)
+				{
+					result[i] = std::sqrt(result[i]);
+				}
+
+				return result;
+			}
+
+		public:
+			SzegoPolynomial(const double & norm_measure, std::array<double, order_> && verblunsky_coefficients) :
+				_norm_measure(norm_measure),
+				_verblunsky_coefficients(verblunsky_coefficients),
+				_norms(_calculate_norms())
+			{
+			}
+
+			SzegoPolynomial(const SzegoPolynomial &) = default;
+			SzegoPolynomial(SzegoPolynomial &&) = default;
+			~SzegoPolynomial() = default;
+			SzegoPolynomial & operator= (SzegoPolynomial &&) = default;
+			SzegoPolynomial & operator= (const SzegoPolynomial &) = default;
+
+			// Evaluate the normalized polynomials at z on the real axis, in the interval [-1, +1].
+			// Note that, contrary to the literature [S:2005A], we use an integral measure dmu, which
```

Comment:
Since you mention this reference can you add it to the references.yaml?

Reply:
I did not find the book anywhere publicly. So I changed there reference to an arXiv paper, which is now added as `[S:2004B]`.